### ISSUE TITLE
Error message: No valid variant remaining

### ISSUE THREAD

**Issue by jpouget** [choishingwan/PRSice#167]
Hello, 

I'm trying to run PRSice using the Angst Anxiety case control data as a base, and my own sample as a target. Would appreciate help troubleshooting an error I have run into, which is also happening for another of the base datasets I'm trying to work with. I am using the command below:
```
Rscript $HOME/scripts/PRSice/PRSice.R --dir . \
    --prsice $HOME/scripts/PRSice/PRSice_mac \
    --base $HOME/data/gwas_summary_stats/anxiety_angst/anxiety.meta.full.cc.tbl.gz \
    --snp 0 --chr 1 --bp 2 --A1 3 --A2 4 --stat 6 --pvalue 8 --index \
    --target $HOME/1_Research/Analyses/gwas/qc/chr1-22.dose.merged_QC_IDupdate_SNPupdate \
    --pheno $HOME/1_Research/Analyses/gwas/assoc/PPDandPPP.pheno \
    --ld $HOME/data/1kg_clean/clean-1KG-noindels \
    --thread 1 \
    --stat BETA \
    --beta \
    --binary-target F \
    --out anx_edps
```

I get the following error:
```
GZ file detected.6330996 variant(s) observed in base file,
with:
1 variant(s) excluded as they are on unknown/sex chromosome
6330995 NA stat/p-value observed
0 total variant(s) included from base file

Error: No valid variant remaining
```

The header format of the base GWAS file is below:
```
SNPID	CHR	BP	Allele1	Allele2	Freq1	Effect	StdErr	P.value	TotalN
rs1	1	226580387	t	g	0.8266	-0.0574	0.0348	0.09867	17310
rs2	1	162736463	t	c	0.8545	0.0654	0.0461	0.1564	10864
rs3	1	222359612	t	c	0.336	-0.0229	0.0308	0.458	17310
```

**Comment by choishingwan** [comment]
Which version are you using? If you don't use --index, and instead provide
the header name, does it run correctly? As of now, it seems like PRSice is
thinking the 9th column (8 from 0 base) contains only of NA. Likely this
column contains special characters that forbid PRSice to convert it into
numeric values.

Sam

**Comment by choishingwan** [comment]
It is likely that your base file contains some special characters that
happens to be removed by R. Good that re-writing the file works for you

**Comment by jpouget** [comment]
Running into the same problem again with the [PGC UKBiobank depression](https://www.nature.com/articles/s41593-018-0326-7) dataset. Again, no non-numeric values appear to be present in the P/BETA columns. 

Header of base GWAS file:
```
MarkerName	A1	A2	Freq	LogOR	StdErrLogOR	P	CHROM	POS
rs1	a	c	0.0596	-5e-04	0.0095	0.9576	7	92383888
rs2	a	g	0.2242	0.0016	0.0051	0.76	12	126890980
rs3	a	g	0.2999	-0.0041	0.0047	0.3865	4	57561647
rs4	a	g	0.5348	-0.0023	0.0043	0.5864	4	85161558
```
Command used:
```
Rscript ~/PRSice.R --dir . \
    --prsice ~/PRSice_mac \
    --base ~/PGC_UKB_depression_genome-wide.rewrite_new.txt \
    --snp MarkerName --chr CHROM --bp POS --A1 A1 --A2 A2 --stat LogOR --pvalue P \
    --target ~/chr1-22.dose.merged_QC_IDupdate_SNPupdate \
    --pheno ~/PPDandPPP.pheno \
    --ld ~/clean-1KG-noindels \
    --thread 1 \
    --stat BETA \
    --beta \
    --binary-target F \
    --out mdd_edps
```

Output:
```
Rscript ~/PRSice.R --dir . \
>     --prsice ~/PRSice_mac \
>     --base ~/PGC_UKB_depression_genome-wide.rewrite_new.txt \
>     --snp MarkerName --chr CHROM --bp POS --A1 A1 --A2 A2 --stat LogOR --pvalue P \
>     --target ~/chr1-22.dose.merged_QC_IDupdate_SNPupdate \
>     --pheno ~/PPDandPPP.pheno \
>     --ld ~/clean-1KG-noindels \
>     --thread 1 \
>     --stat BETA \
>     --beta \
>     --binary-target F \
>     --out mdd_edps
PRSice 2.2.11.b (2019-10-16)
https://github.com/choishingwan/PRSice
(C) 2016-2019 Shing Wan (Sam) Choi and Paul F. O'Reilly
GNU General Public License v3
If you use PRSice in any published work, please cite:
Choi SW, O'Reilly PF.
PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.
GigaScience 8, no. 7 (July 1, 2019)
2019-11-25 08:34:37
~/PRSice_mac \
    --A1 A1 \
    --A2 A2 \
    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \
    --base ~/PGC_UKB_depression_genome-wide.rewrite_new.txt \
    --beta  \
    --binary-target F \
    --bp POS \
    --chr CHROM \
    --clump-kb 250 \
    --clump-p 1.000000 \
    --clump-r2 0.100000 \
    --interval 5e-05 \
    --ld ~/clean-1KG-noindels \
    --lower 5e-08 \
    --out mdd_edps \
    --pheno ~/PPDandPPP.pheno \
    --pvalue P \
    --seed 1853377881 \
    --snp MarkerName \
    --stat BETA \
    --target ~/chr1-22.dose.merged_QC_IDupdate_SNPupdate \
    --thread 1 \
    --upper 0.5

Initializing Genotype file:
~/chr1-22.dose.merged_QC_IDupdate_SNPupdate
(bed)

Start processing PGC_UKB_depression_genome-wide.rewrite_new
==================================================

Reading 100.00%
Base file:
~/PGC_UKB_depression_genome-wide.rewrite_new.txt
8102277 variant(s) observed in base file, with:
11411 duplicated variant(s)
8090866 NA stat/p-value observed
0 total variant(s) included from base file

Error: No valid variant remaining

Error:
Execution halted
```

Rewriting the file in R did not solve the error for this dataset. 

Wondering what else I can do to troubleshoot?

**Comment by choishingwan** [comment]
Unfortunately, with this situation, I might need to have a look at your
summary statistic file. Is that sharable? If not, if you open the file in a
command line text editor such as vi, do you see some funny symbol behind
your p-value column?


Dr Shing Wan Choi
Postdoctoral Fellow
Genetics and Genomic Sciences
Icahn School of Medicine, Mount Sinai, NYC

**Comment by choishingwan** [comment]
Maybe try this build (2.2.12
<https://www.dropbox.com/s/51ydin4h7mktfmb/PRSice_linux?dl=0>)? I have put
in a more aggressive trimming function which should automatically remove
special characters if any.
Dr Shing Wan Choi
Postdoctoral Fellow
Genetics and Genomic Sciences
Icahn School of Medicine, Mount Sinai, NYC

**Comment by nancyya** [comment]
Hi, I'm having the same problem. I'm running on Linux, with the latest PRSice version for linux.
I can't share the target data but I use the publicly available base data from the "GWAS catalog": [here](ftp://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/NicolasA_29566793_GCST005647) and [here](ftp://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/vanRheenenW_27455348_GCST004692)

I run the following command: 
`Rscript PRSice.R --prsice /home/PRSice/PRSice_linux \
--base /home/PRSice/base_data/ALS/vanRheenen/27455348-GCST004692-EFO_0000253.h.tsv.gz \ 
--snp hm_rsid --chr hm_chrom --bp hm_pos --A1 hm_effect_allele --A2 hm_other_allele --stat hm_beta --beta --pvalue p_value \
--target /home/ProjectMiNE/WGS_datafreeze1/WGS_datafreeze1/WGS_datafreeze1_chr#_maf_0.0001,/home/ProjectMiNE/WGS_datafreeze1/WGS_datafreeze1/WGS_datafreeze1_maf_0.0001.fam \
--binary-target T \
--out PRSice_ALS_vanRheenen_DF1`


The log:

```
Start processing 27455348-GCST004692-EFO_0000253.h.tsv 
================================================== 

Due to library restrictions, we cannot display progress bar 
for gz 

Reading 100.00%
Base file: 
/home/PRSice/base_data/ALS/vanRheenen/27455348-GCST004692-EFO_0000253.h.tsv.gz 
GZ file detected. Header of file is: 

hm_variant_id	hm_rsid	hm_chrom	hm_pos	hm_other_allele	hm_effect_allele	hm_beta	hm_odds_ratio	hm_ci_lower	hm_ci_upper	hm_effect_allele_frequency	hm_code	chromosome	vari nt_id	base_pair_location	effect_allele	other_allele	freq	beta	standard_error	p_value	cases	controls	effect_allele_frequency	ci_upper	ci_lower	odds_ratio
8704844 variant(s) observed in base file, with: 
48549 duplicated variant(s) 
1 variant(s) excluded as they are on unknown/sex chromosome 
12682 NA stat/p-value observed 
1235666 ambiguous variant(s) excluded 
7407946 total variant(s) included from base file 

Loading Genotype info from target 
================================================== 

6197 people (3632 male(s), 2544 female(s)) observed 
6197 founder(s) included 

38539664 variant(s) not found in previous data 
0 variant(s) included 

Error: No vairant remained! 

Error: 
Execution halted
```

**Comment by choishingwan** [comment]
Sorry, can't remember if I have replied to it in other platform. If your variant id is matched between the WGS and target data, then PRSice should work. However, judging from the error message, it does seems like they don't. I suggest you to construct the variant ID using chr:position without including the effective and non-effective allele just in case flipping is required between the base and target. 

For now, I don't really want to implement the automatic chr:bp mapping as we believe this should be a user decision. Though I might, when I have completed the unit tests (which will take a long time), implement this feature in the future.
